pages/treeherder.py

```diff
@@ -378,38 +373,50 @@ def title(self):
 
             def click(self):
                 self.root.click()
-                self.wait.until(lambda _: self.page.job_details.job_result_status)
+                self.wait.until(lambda _: self.page.info_panel.job_details.job_result_status)
 
-    class JobDetails(Region):
+    class InfoPanel(Region):
 
-        _job_details_panel_locator = (By.ID, 'job-details-panel')
-        _job_keyword_locator = (By.CSS_SELECTOR, '#job-details-pane > ul > li > a:nth-last-child(1)')
-        _job_result_status_locator = (By.CSS_SELECTOR, '#result-status-pane > div:nth-child(1) > span:nth-child(2)')
-        _logviewer_button_locator = (By.ID, 'logviewer-btn')
-        _pin_job_locator = (By.ID, 'pin-job-btn')
+        _root_locator = (By.ID, 'info-panel-content')
 
         @property
-        def job_keyword_name(self):
-            return self.find_element(*self._job_keyword_locator).text
+        def is_open(self):
+            return self.root.is_displayed()
 
         @property
-        def job_result_status(self):
-            return self.find_element(*self._job_result_status_locator).text
-
-        def filter_by_job_keyword(self):
-            self.find_element(*self._job_keyword_locator).click()
-
-        def open_logviewer(self):
-            self.find_element(*self._job_details_panel_locator).send_keys('l')
-            window_handles = self.selenium.window_handles
-            for handle in window_handles:
-                self.selenium.switch_to.window(handle)
-            return LogviewerPage(self.selenium, self.page.base_url).wait_for_page_to_load()
-
-        def pin_job(self):
-            el = self.find_element(*self._job_keyword_locator)
-            self.wait.until(EC.visibility_of(el))
-            self.find_element(*self._pin_job_locator).click()
+        def job_details(self):
+            return self.JobDetails(self)
```

Comment:
The `Region` constructor expects a `Page` and `self` is a `Region`. Change this to `return self.JobDetails(self.page)` and your failure should go away.